core/src/server/rpc_module.rs

```diff
@@ -764,25 +764,29 @@ impl SubscriptionSink {
 	/// when items gets produced by the stream.
 	///
 	/// Returns `Ok(())` if the stream or connection was terminated.
-	/// Returns `Err(_)` if one of the items couldn't be serialized.
+	/// Returns `Err(_)` immediately if the underlying stream returns an error or if an item from the stream could not be serialized.
 	///
 	/// # Examples
 	///
 	/// ```no_run
 	///
 	/// use jsonrpsee_core::server::rpc_module::RpcModule;
+	/// use anyhow::anyhow;
 	///
 	/// let mut m = RpcModule::new(());
 	/// m.register_subscription("sub", "_", "unsub", |params, mut sink, _| {
-	///     let stream = futures_util::stream::iter(vec![1_u32, 2, 3]);
+	///     let stream = futures_util::stream::iter(vec![Ok(1_u32), Ok(2), Err("error on the stream")]);
+	///     // This will return send `[Ok(1_u32), Ok(2_u32)]` to the subscriber
+	///     // because after the `Err(_)` the stream is terminated.
 	///     tokio::spawn(sink.pipe_from_stream(stream));
 	///     Ok(())
 	/// });
 	/// ```
-	pub async fn pipe_from_stream<S, T>(mut self, mut stream: S) -> Result<(), Error>
+	pub async fn pipe_from_stream<S, T, E>(mut self, mut stream: S) -> Result<(), Error>
```

Comment:
It definitely makes sense to me to support streams of results (so that we can handle the errors better and such)!

I wonder whether it's worth supporting both `pipe_from_stream` and `pipe_result_from_stream` or something (the former just converting everything to `Ok(..)` and sending it on to `pipe_result_from_stream`) just for the added convenience?

Reply:
good idea, yepp I like that